src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/BinaryEncoders/MessageBinaryEncoderBase.cs

```diff
@@ -96,6 +96,20 @@ protected bool IsEncryptionConfigured
             }
         }
 
+        /// <summary>
+        /// [Beta] Gets the maximum size of the encrypted document.
+        /// </summary>
+        /// <value>
+        /// The maximum size of the encrypted document.
+        /// </value>
+        protected int? MaxEncryptionDocumentSize
```

Comment:
I don't this name reflects the semantics of the new setting.  We want to allow each document in the array of writes to be MaxDocumentSize. The value of this doesn't change.  But we also want to split if the total size of the array of writes exceeds 2 MiB (with the exception of the first write).

Reply:
This is a new constant... It just has a funny name.  I called it `maxSplittableDocumentSize` in the Java driver.  That's a bit more accurate but it's not completely descriptive either.